### DISCUSSION TITLE
[IDEA] Make a bridge to the PHP-FPM

### DISCUSSION THREAD

**Discussion by rustatian** [roadrunner-server/roadrunner#753] [Ideas]
After discussion with @wolfy-j .
Support a PHP-FPM (https://www.php.net/manual/en/install.fpm.php) process manager. Since we have the ability to redirect any input from any plugin (which uses workers and `Pool` implementation) to any external system, we can design and implement a new `Pool` that will redirect input to the PHP-FPM workers instead of RR workers.
TODO:
1. Implement `Pool` interface, create a new dir inside the `pkg` folder named `fpm`.
2. Add a new configuration section, to have the ability to use `php-fpm` pool impl.
3. We can use `server` plugin and implement a new, let's say, relay.
 
Useful links:
https://www.nginx.com/resources/wiki/start/topics/examples/phpfcgi/
https://github.com/php/php-src/tree/master/sapi/fpm/fpm

**Comment by tarampampam** [comment]
@rustatian Often `nginx` is not only reverse proxy before the app and static assets - it is used for common HTTP header management (cache-control), proxying requests to other upstream, redirects, etc. So, if RoadRunner will be used for an Nginx replacement - we should implement a lot of Nginx features in RoadRunner. So, we really want it?

**Comment by rustatian** [comment]
@tarampampam Yeah, I am pretty sure, that we shouldn't re-implement `nginx` here 😄 Ok, so, if the community requested a bridge like that, we will continue the discussion.
